### Eclipse Workspace Patch 1.0
#P estudiolivre_head
Index: src/lib/elgal/model/FileReference.php
===================================================================
RCS file: /cvsroot/arca/estudiolivre/src/lib/elgal/model/FileReference.php,v
retrieving revision 1.17
diff -u -r1.17 FileReference.php
--- lib/elgal/model/FileReference.php	30 Jul 2007 22:12:22 -0000	1.17
+++ lib/elgal/model/FileReference.php	9 Aug 2007 21:48:16 -0000
@@ -58,6 +58,16 @@
 
  	}
 	
+	function getClassName($lcName) {
+		$classes = array("audiofile" => "AudioFile",
+						 "videofile" => "VideoFile",
+						 "imagefile" => "ImageFile",
+						 "textfile" => "TextFile",
+						 "plainfile" => "PlainFile",
+						 "zipfile" => "ZipFile");
+		return $classes[$lcName];
+	}
+	
 	function hitDownload() {
 		return $this->update(array('downloads' => $this->downloads+1));
 	}
Index: src/lib/elgal/model/Publication.php
===================================================================
RCS file: /cvsroot/arca/estudiolivre/src/lib/elgal/model/Publication.php,v
retrieving revision 1.13
diff -u -r1.13 Publication.php
--- lib/elgal/model/Publication.php	25 Jul 2007 23:58:02 -0000	1.13
+++ lib/elgal/model/Publication.php	9 Aug 2007 21:48:16 -0000
@@ -44,6 +44,15 @@
 	var $tagType = "gallery";
 	/*************************************************************/
 
+	function getClassName($lcName) {
+		$classes = array("audiopublication" => "AudioPublication",
+						 "videopublication" => "VideoPublication",
+						 "imagepublication" => "ImagePublication",
+						 "textpublication" => "TextPublication",
+						 "otherpublication" => "OtherPublication");
+		return $classes[$lcName];
+	}
+
 	function checkRequiredField($value, $msg) {
 		if (preg_match('/^\s*$/',$value)) {
 			return $msg;
