Index: lib/persistentObj/PersistentObject.php
===================================================================
--- lib/persistentObj/PersistentObject.php	(revision 55)
+++ lib/persistentObj/PersistentObject.php	(working copy)
@@ -50,7 +50,7 @@
 	    if (is_array($fields)) {
 	    	if (count($fields)) {
 	    		if (isset($this->actualClass))
-			    	$fields['actualClass'] = get_class($this);
+			    	$fields['actualClass'] = $this->getClassName(get_class($this));
 		    	$this->_populateObject($fields);
 		    	$this->id = $this->insert($fields);
 	    		$this->_insertIndex();
@@ -65,6 +65,15 @@
 	    return $this;
 	}
 
+	/* Another PHP4 hack, since get_class returns lowercase
+	 * in this class it just returns. any superclass that has
+	 * subclasses that will be instaciated dinamicaly should 
+	 * re-implement this method correctly
+	 */
+	function getClassName($name) {
+		return $name;
+	}
+
     /*
      * hash getFields
      */
