--- tiki-editpage.php	2006-04-24 08:26:49.000000000 -0300
+++ tiki-editpage.php	2006-05-08 19:39:59.000000000 -0300
@@ -668,6 +668,7 @@
   $cat_href="tiki-index.php?page=".urlencode($cat_objid);
   include_once("categorize.php");
   include_once("poll_categorize.php");
+  include_once("freetag_apply.php");
 
   if ((($feature_wiki_description == 'y')
     && (md5($info["description"]) != md5($_REQUEST["description"])))
@@ -808,6 +809,7 @@
 $cat_type = 'wiki page';
 $cat_objid = $_REQUEST["page"];
 include_once ("categorize_list.php");
+include_once ("freetag_list.php");
 
 if ($feature_categories == 'y' && isset($_SERVER['HTTP_REFERER']) && strstr($_SERVER['HTTP_REFERER'], 'tiki-index.php') && !$tikilib->page_exists($_REQUEST["page"])) { // default the categs the page you come from for a new page
 	if (preg_match('/page=([^\&]+)/', $_SERVER['HTTP_REFERER'], $ms))
--- templates/tiki-editpage.tpl	2007-11-12 17:56:45.000000000 -0200
+++ templates/tiki-editpage.tpl	2007-11-12 17:55:04.000000000 -0200
@@ -269,6 +269,9 @@
 </td></tr>
 {/if}
 {/if}
+{if $feature_freetags eq 'y' and $tiki_p_freetags_tag eq 'y'}
+{include file=freetag.tpl}
+{/if}
 {if $feature_wiki_allowhtml eq 'y' and $tiki_p_use_HTML eq 'y' and $page|lower neq 'sandbox'}
 <tr class="formcolor"><td>{tr}Allow HTML{/tr}: </td><td><input type="checkbox" name="allowhtml" {if $allowhtml eq 'y'}checked="checked"{/if}/></td></tr>
 {/if}
