--- lib/searchlib.php	2006-05-09 17:05:19.000000000 -0300
+++ lib/searchlib.php	2006-05-09 17:30:56.389972992 -0300
@@ -158,7 +158,8 @@
 
 		if ($chkCatPerm) {
 
-		    $sqlJoin .= " LEFT JOIN `tiki_categorized_objects` co ON co.`type`=? AND co.`objId`=$objKeyCat ";
+		    $sqlJoin .= " LEFT JOIN `tiki_objects` obj ON obj.`type`=? AND obj.`objectId`=$objKeyCat ";
+		    $sqlJoin .= " LEFT JOIN `tiki_categorized_objects` co ON co.`catObjectId`=obj.`objectId` ";
 		    $sqlJoin .= " LEFT JOIN `tiki_category_objects` cat ON co.`catObjectId`=cat.`catObjectId` ";
 
 		    $bindJoin[] = $objType;
@@ -170,7 +171,7 @@
 		    }
                     if( $forbiddenCatStr == "" ) $forbiddenCatStr = '\'\'';
 
-		    $sqlFields .= ', co.`objId` IS NOT NULL as categorized, MAX(cat.`categId` IN ('.$forbiddenCatStr.')) as forbidden ';
+		    $sqlFields .= ', co.`catObjectId` IS NOT NULL as categorized, MAX(cat.`categId` IN ('.$forbiddenCatStr.')) as forbidden ';
 		    $bindFields = array_merge($bindFields, $forbiddenCatList);
 
 		    $sqlGroup = " GROUP BY $objKeyGroup ";
