--- tiki-index.php	2006-04-24 08:26:50.000000000 -0300
+++ tiki-index.php	2006-05-08 19:41:13.000000000 -0300
@@ -12,6 +12,10 @@
 include_once('lib/wiki/wikilib.php');
 include_once('lib/stats/statslib.php');
 
+if ($feature_freetags == 'y') {
+	include_once('lib/freetag/freetaglib.php');
+}
+
 if ($feature_categories == 'y') {
 	global $categlib;
 	if (!is_object($categlib)) {
@@ -574,6 +578,10 @@
     HAWTIKI_index($info);
 }
 
+if ($feature_freetags == 'y') {     // And get the Tags for the posts
+    $tags = $freetaglib->get_tags_on_object($info['pageName'], 'wiki page');
+    $smarty->assign('freetags',$tags);
+}
 // Display category path or not (like {catpath()})
 if (isset($is_categorized) && $is_categorized) {
     $smarty->assign('is_categorized','y');
