--- tiki-setup.php	2006-05-12 16:52:58.000000000 -0300
+++ tiki-setup.php	2006-07-21 12:37:46.000000000 -0300
@@ -1708,9 +1708,6 @@
 }
 $smarty->assign('page', $page);

-// We include Tiki modules here (any idea why right here ?)
-include_once ("tiki-modules.php");
-
 if ($feature_warn_on_edit == 'y') {

     if (strstr($_SERVER['REQUEST_URI'], 'tiki-editpage')) {
--- setup_smarty.php	2006-05-12 16:52:56.000000000 -0300
+++ setup_smarty.php	2006-07-21 13:40:26.000000000 -0300
@@ -153,6 +165,22 @@
   		}
 		return $this->template_dir.$file.$template;
 	}
+	function display($resource_name, $cache_id=null, $compile_id = null) {
+	    global $feature_ajax, $ajaxlib;
+		if ($feature_ajax == 'y') {
+		    $ajaxlib->processRequests();
+		}
+		if ($resource_name == 'confirm.tpl' || $resource_name == 'error.tpl' || $resource_name == 'information.tpl' || $resource_name == 'error_ticket.tpl' || $resource_name == 'error_simple.tpl') {
+			include_once('tiki-modules.php');
+		} elseif (($tpl = $this->get_template_vars('mid')) && $resource_name == 'tiki.tpl') {
+			global $currentTpl;
+			$currentTpl = $tpl;
+			$data = $this->fetch($tpl, $cache_id, $compile_id);//must get the mid because the modules can overwrite smarty variables
+			$this->assign('mid_data', $data);
+			include_once('tiki-modules.php');
+		}
+		return parent::display($resource_name, $cache_id, $compile_id);
+	}
 }
 
 $smarty = new Smarty_TikiWiki($tikidomain);
