--- tiki-editpage.php	2007-12-05 05:51:00.000000000 -0200
+++ tiki-editpage.php	2008-03-18 12:46:37.000000000 -0300
@@ -156,17 +156,21 @@
 if ($tikidomain) { $wiki_up.= "/$tikidomain"; }
 // Upload pictures here
 if (($feature_wiki_pictures == 'y') && (isset($tiki_p_upload_picture)) && ($tiki_p_upload_picture == 'y')) {
-  if (isset($_FILES['picfile1']) && is_uploaded_file($_FILES['picfile1']['tmp_name'])) {
-    $picname = $_FILES['picfile1']['name'];
-
-		if (preg_match('/\.(gif|png|jpe?g)$/i',$picname)) { 
-			if (@getimagesize($_FILES['picfile1']['tmp_name'])) {
-				move_uploaded_file($_FILES['picfile1']['tmp_name'], "$wiki_up/$picname");
-				chmod("$wiki_up/$picname", 0644); // seems necessary on some system (see move_uploaded_file doc on php.net
-			}
-		}
-    //is done in js... $_REQUEST["edit"] = $_REQUEST["edit"] . "{img src=\"img/wiki_up/$tikidomain$picname\"}";
-  }
+    if (isset($_REQUEST['img_form_count'])) {
+	for ($i = 1; $i <= $_REQUEST['img_form_count']; $i++) {
+	    if (isset($_FILES["picfile" . $i]) && is_uploaded_file($_FILES["picfile" . $i]['tmp_name'])) {
+		$picname = $_FILES["picfile" . $i]['name'];
+		
+		if (preg_match('/\.(gif|png|jpe?g)$/i',$picname)) {
+		    if (@getimagesize($_FILES['picfile' . $i]['tmp_name'])) {
+		        move_uploaded_file($_FILES["picfile" . $i]['tmp_name'], "$wiki_up/$picname");
+		        chmod("$wiki_up/$picname", 0644); // seems necessary on some system (see move_uploaded_file doc on php.net
+		    }
+ 		}
+		//is done in js... $_REQUEST["edit"] = $_REQUEST["edit"] . "{img src=\"img/wiki_up/$tikidomain$picname\"}";
+	    }
+	}
+    } 
 }
 
 /**
