--- tiki-index.php	2006-04-24 08:26:50.000000000 -0300
+++ tiki-index.php	2006-05-16 10:16:58.000000000 -0300
@@ -533,6 +537,19 @@
     include('tiki-tc.php');
 }
 
+// ESTUDIOLIVRE HACK BEGIN - associa categoria ao smarty, pra fazer theme control na mao
+$tc_categs = $categlib->get_object_categories($cat_type, $cat_objid);
+
+if (count($tc_categs)) {
+    foreach ($tc_categs as $cat) {
+	$catt=$categlib->get_category($cat);
+	$smarty->assign_by_ref('category', $catt["name"]);
+	break;
+    }
+}
+// ESTUDIOLIVRE HACK END
+
+
 // Watches
 if($feature_user_watches == 'y') {
     if($user && isset($_REQUEST['watch_event'])) {
