--- lib/init/tra.php	2007-11-13 00:00:48.000000000 -0200
+++ lib/init/tra.php	2007-11-12 23:58:22.000000000 -0200
@@ -26,6 +26,9 @@
 					if (is_file("lang/$l/custom.php")) {
 						include_once("lang/$l/custom.php");
 					}
+					if (file_exists("lang/$l/el-additional_strings.php")) {
+					    include_once("lang/$l/el-additional_strings.php");
+					}
 				}
 			if (isset($lang[$content])) {
 				return $lang[$content];
--- setup_smarty.php	2006-05-12 16:52:56.000000000 -0300
+++ setup_smarty.php	2006-07-11 18:36:07.000000000 -0300
@@ -95,6 +103,8 @@
 			$lgSave = $language;
 			$language = $lg;
 			include("lang/$language/language.php");
+			if (file_exists("lang/$language/el-additional_strings.php"))
+			    include_once("lang/$language/el-additional_strings.php");
 				// the language file needs to be included again:
 				// the file could have been included before: prefilter.tr using include_once will not reload the file
 				// but the $lang can be from another language
@@ -103,6 +113,8 @@
 		if (!$isCompiled) {
 			$language = $lgSave;
 			include ("lang/$language/language.php");
+			if (file_exists("lang/$language/el-additional_strings.php"))
+			    include_once("lang/$language/el-additional_strings.php");
 		}
 
 		return ereg_replace("^[ \t]*", "", $res);
