--- modules/mod-breadcrumb.php	2004-08-26 14:39:15.000000000 -0300
+++ modules/mod-breadcrumb.php	2006-07-20 21:59:06.000000000 -0300
@@ -9,11 +9,11 @@
   exit;
 }
 
-if (!isset($_SESSION["breadCrumb"])) {
-	$_SESSION["breadCrumb"] = array();
+if (!isset($_SESSION["elBreadCrumbs"])) {
+	$_SESSION["elBreadCrumbs"] = array();
 }
 
-$bbreadCrumb = array_reverse($_SESSION["breadCrumb"]);
-$smarty->assign('breadCrumb', $bbreadCrumb);
+$bbreadCrumb = $_SESSION["elBreadCrumbs"];
+$smarty->assign('elCrumbs', $bbreadCrumb);
 $smarty->assign('maxlen', isset($module_params["maxlen"]) ? $module_params["maxlen"] : 0);
 ?>
