--- lib/feedcreator/feedcreator.class.php	2006-08-15 17:14:42.000000000 -0300
+++ lib/feedcreator/feedcreator.class.php	2006-08-15 17:47:22.000000000 -0300
@@ -183,7 +183,7 @@
 	/**
 	 * Optional attributes of an item.
 	 */
-	var $author, $authorEmail, $image, $category, $comments, $guid, $source, $creator;
+	var $author, $authorEmail, $image, $category, $comments, $guid, $source, $creator, $enclosure;
 	
 	/**
 	 * Publishing date of an item. May be in one of the following formats:
@@ -996,6 +996,9 @@
 			if ($this->items[$i]->guid!="") {
 				$feed.= "            <guid>".htmlspecialchars($this->items[$i]->guid)."</guid>\n";
 			}
+			if ($this->items[$i]->enclosure && is_array($this->items[$i]->enclosure)) {
+			    $feed.= '            <enclosure url="'.htmlspecialchars($this->items[$i]->enclosure["url"]).'" lenght="'.htmlspecialchars($this->items[$i]->enclosure["lenght"]).'" type="'.htmlspecialchars($this->items[$i]->enclosure["type"]).'"/>';
+			}
 			$feed.= $this->_createAdditionalElements($this->items[$i]->additionalElements, "        ");
 			$feed.= "        </item>\n";
 		}
--- lib/rss/rsslib.php	2006-08-15 16:46:37.000000000 -0300
+++ lib/rss/rsslib.php	2006-08-15 17:41:39.000000000 -0300
@@ -199,7 +199,7 @@
 
 		$url = htmlspecialchars($this->httpPrefix().$_SERVER["REQUEST_URI"]);
 		$home = htmlspecialchars($this->httpPrefix().dirname( $urlarray["path"] ).$dirname.$tikiIndex);
-		$img = htmlspecialchars($this->httpPrefix().dirname( $urlarray["path"] ).$dirname."img/tiki.jpg");
+		$img = htmlspecialchars($this->httpPrefix().dirname( $urlarray["path"] ).$dirname."styles/bolha/img/logoTop.png");
 
 		$title = htmlspecialchars($title);
 		$desc = htmlspecialchars($desc);
@@ -267,7 +267,7 @@
 		$rss->feedURL = $url;
 		
 		$image = new FeedImage();
-		$image->title = tra("tikiwiki logo");
+		$image->title = tra("EstudioLivre");
 		$image->url = $img;
 		$image->link = $home;
 		$image->description = tra(sprintf("Feed provided by %s. Click to visit.", $home));
@@ -323,8 +323,12 @@
 					if ($tmp<>"") $item->author = $tmp;
 				} else $item->author = $data["$authorId"];
 			}
-			 
-			$rss->addItem($item); 
+			
+			if ($data["enclosure"]) {
+			    $item->enclosure = $data["enclosure"];
+			}
+
+			$rss->addItem($item);
 		} 
 		$data = $rss->createFeed($this->get_rss_version_name($rss_version));
 		$this->put_to_cache($uniqueid, (int) $data, $rss_version);
