--- setup_smarty.php	2006-06-15 14:21:59.000000000 -0300
+++ setup_smarty.php	2006-06-15 14:19:58.000000000 -0300
@@ -43,13 +43,24 @@
 	function _smarty_include($params) {
 		global $style, $style_base, $tikidomain;
 
+		$isIe = preg_match('/MSIE/', $_SERVER['HTTP_USER_AGENT']) && 
+ 		    !preg_match('/Opera/', $_SERVER['HTTP_USER_AGENT']);
+
 		if (isset($style) && isset($style_base)) {
 			if ($tikidomain and file_exists("templates/$tikidomain/styles/$style_base/".$params['smarty_include_tpl_file'])) {
 				$params['smarty_include_tpl_file'] = "$tikidomain/styles/$style_base/".$params['smarty_include_tpl_file'];
 			} elseif ($tikidomain and file_exists("templates/$tikidomain/".$params['smarty_include_tpl_file'])) {
 				$params['smarty_include_tpl_file'] = "$tikidomain/".$params['smarty_include_tpl_file'];
 			} elseif (file_exists("templates/styles/$style_base/".$params['smarty_include_tpl_file'])) {
-				$params['smarty_include_tpl_file'] = "styles/$style_base/".$params['smarty_include_tpl_file'];
+			    if ($isIe && file_exists("templates/styles/$style_base/ie-".$params['smarty_include_tpl_file'])) {
+				$params['smarty_include_tpl_file'] = 'ie-' . $params['smarty_include_tpl_file'];
+			    }
+			    $params['smarty_include_tpl_file'] = "styles/$style_base/".$params['smarty_include_tpl_file'];
+			} elseif ($isIe && file_exists("templates/ie-" . $params['smarty_include_tpl_file'])) {
+			    $params['smarty_include_tpl_file'] = 'ie-' . $params['smarty_include_tpl_file'];
 			}
 		}
+		global $currentTpl;
+		$currentTpl = $params['smarty_include_tpl_file'];
+
 		return parent::_smarty_include($params);
