--- lib/smarty_tiki/modifier.avatarize.php	2007-11-12 18:02:10.000000000 -0200
+++ lib/smarty_tiki/modifier.avatarize.php	2007-11-12 18:20:36.000000000 -0200
@@ -19,7 +19,7 @@
   global $tikilib;
   $avatar = $tikilib->get_user_avatar($user);
   if ($tikilib->get_user_preference($user,'user_information','public')=='public') {
-  	$avatar = "<a title='$user' href='tiki-user_information.php?view_user=$user'>".$avatar.'</a>';
+      $avatar = "<a title='$user' href='el-user.php?view_user=$user'>".$avatar.'</a>'; 
   } 
   return $avatar;	
 }
