--- tiki-view_forum_thread.php	2007-07-25 15:05:02.000000000 -0300
+++ tiki-view_forum_thread.php	2007-07-25 15:03:16.000000000 -0300
@@ -296,6 +296,14 @@
     $smarty->assign('reported', $commentslib->get_num_reported($_REQUEST['forumId']));
 }
 
+if ($feature_freetags == 'y') {
+    require_once("lib/freetag/freetaglib.php");
+    $cat_type = 'forum post';
+    $cat_objid = $comments_parentId;
+    $tags = $freetaglib->get_tags_on_object($cat_objid, $cat_type);
+    $smarty->assign('freetags',$tags);
+}
+
 include_once("textareasize.php");
 
 if ($feature_forum_parse == "y") {
