--- lib/tikilib.php.orig	2006-11-24 15:02:40.000000000 -0200
+++ lib/tikilib.php	2006-11-24 15:34:34.000000000 -0200
@@ -3615,7 +3615,7 @@
 	$res = $result->fetchRow();
 
 	if (!$res["description"])
-	    $res["description"] = $pageName;
+	    $res["description"] = tra('no description');
 
 	return $res["description"];
     }
@@ -5094,7 +5094,7 @@
 				$linktext = $pages[1][$i];
 			}
 
-		    $repl = '<a title="'.$desc1.'" href="'.$uri_ref.'" class="wiki">' . $linktext . '</a>';
+		    $repl = '<a href="'.$uri_ref.'" class="wiki">' . $linktext . '</a>';
 
 		    // Check is timeout expired?
 		    if (isset($text[1]) && (time() - intval($this->page_exists_modtime($pages[1][$i]))) < intval($text[1]))
