--- tiki-setup.php	2007-06-06 20:37:18.000000000 -0300
+++ tiki-setup.php	2007-06-06 20:39:56.000000000 -0300
@@ -139,9 +139,14 @@
 
             print "
 <html><body>
+<center><img src=\"styles/bolha/img/logoTop.png\">
+<h2>Estamos em manuten&ccedil;&atilde;o</h2>
+<h3>Volte mais tarde</h3></center>
+<!-- 
 <h2><font color='red'>Tikiwiki is not properly set up:</font></h1>
-<pre>
 $errors
+-->
+</body></html>
 ";
 						if ($tikidomain) {
 							$install_link = '?multi='.urlencode($tikidomain);
