--- tiki-setup_base.php	2007-06-06 18:05:30.000000000 -0300
+++ tiki-setup_base.php	2007-06-06 18:05:48.000000000 -0300
@@ -177,8 +177,10 @@
 // is not yet set up and give a message
 $rc = @$tikilib->getOne( "select `value` from `tiki_preferences` where `name`=?",array("language"),false);
 if (!$rc) {
-	print "<html><head><title>Site not available</title></head><body><center><br />";
-	print "<b>Notice: this site is currently unavailable.<br /><br />Please try again later or contact the admin.</b></center></body></html>";
+    print "<html><body>
+<center><img src=\"styles/bolha/img/logoTop.png\">
+<h2>Estamos em manuten&ccedil;&atilde;o</h2>
+<h3>Volte mais tarde</h3></center>";
 	die();
 }
 
