--- ../estudiolivre_base/lib/tikilib.php	2006-05-12 16:53:06.000000000 -0300
+++ lib/tikilib.php	2006-07-14 14:02:32.000000000 -0300
@@ -4480,6 +4481,9 @@
 	    }
 	}
 
+	// ESTUDIOLIVRE HACK - wiki template
+	$this->getWikiTemplate($data);
+
 	$this->parse_first($data, $preparsed, $noparsed);
 
 	// Extract [link] sections (to be re-inserted later)
@@ -5541,6 +5559,42 @@
 	return $data;
     }
 
+		    
+    function getWikiTemplate(&$data) {
+	global $templateStack;      
+	$template = array();
+	$matches = array();
+	
+	preg_match_all("/(?s)\{\{(.*?)(?:\|\|(.*?))?\}\}/", $data, $template);
+	
+	if($template) {      
+	    
+	    for($j=0; $j<sizeof($template[0]); $j++) {
+		
+		if(isset($templateStack[$template[1][$j]]) && $templateStack[$template[1][$j]]) {
+		    global $smarty;
+		    $smarty->assign('msg', tra("Você tentou inserir um template dentro dele mesmo, isso causaria um loop infinito. Por favor conserte isso na edição da página."));
+		    $smarty->display("error.tpl");
+		    die;
+		}
+		
+		$templateStack[$template[1][$j]] = true;
+		$info = $this->get_page_info($template[1][$j]);
+		$pdata = $info["data"];
+		$this->getWikiTemplate($pdata);
+		$templateStack[$template[1][$j]] = false;
+		if(isset($template[2][$j])){
+		    preg_match_all("/(?s)(.+?)\=(.+?)(?:\|\||$)/", $template[2][$j], $matches);
+		    
+		    for($i=0; $i<sizeof($matches[0]); $i++){
+			$pdata = preg_replace("/" . trim($matches[1][$i]) . "/", trim($matches[2][$i]), $pdata);
+		    }
+		}
+		$data = preg_replace("/(?s)\{\{(.*?)\}\}/", trim($pdata), $data, 1);
+	    }
+	}
+    }
+    
     function parse_smileys($data) {
 	global $feature_smileys;
 
