--- ../estudiolivre_base/lib/tikilib.php	2006-05-12 16:53:06.000000000 -0300
+++ lib/tikilib.php	2006-06-09 19:10:47.000000000 -0300
@@ -3528,12 +3528,13 @@
 	if (!isset($user_preferences[$user][$name])) {
 	    $query = "select `value` from `tiki_user_preferences` where `prefName`=? and `user`=?";
 
-	    $result = $this->getOne($query, array( "$name", "$user"));
+	    $result = $this->query($query, array($name, $user));
+		$row = $result->fetchRow();
 
-	    if (!$result) {
+	    if (!is_array($row)) {
 		$user_preferences[$user][$name] = $default;
 	    } else {
-		$user_preferences[$user][$name] = $result;
+		$user_preferences[$user][$name] = $row['value'];
 	    }
 	}
 
