--- tiki-view_forum.php	2007-07-25 15:05:02.000000000 -0300
+++ tiki-view_forum.php	2007-07-25 15:03:01.000000000 -0300
@@ -16,6 +16,10 @@
 	}
 }
 
+if ($feature_freetags == 'y') {
+    include_once('lib/freetag/freetaglib.php');
+}
+
 if ($feature_forums != 'y') {
     $smarty->assign('msg', tra("This feature is disabled").": feature_forums");
 
@@ -333,6 +337,15 @@
 					);
 
 			    // The thread *WAS* successfully created.
+
+			    // TAG Stuff
+			    $cat_type = 'forum post';
+			    $cat_objid = $threadId;
+			    $cat_desc = substr($_REQUEST["comments_data"],0,200);
+			    $cat_name = $_REQUEST["comments_title"];
+			    $cat_href="tiki-view_forum_thread.php?comments_parentId=" . $threadId . "&forumId=" . $_REQUEST["forumId"];
+			    include_once ("freetag_apply.php");
+			    
 			    if( $threadId ) 
 			    {
 				// Deal with mail notifications.
@@ -358,6 +371,14 @@
 		    } elseif ($tiki_p_admin_forum == 'y' || $commentslib->user_can_edit_post($user, $_REQUEST["comments_threadId"])) {
 			    $commentslib->update_comment($_REQUEST["comments_threadId"], $_REQUEST["comments_title"], '', ($_REQUEST["comments_data"]), $_REQUEST["comment_topictype"], $_REQUEST['comment_topicsummary'], $_REQUEST['comment_topicsmiley']);
 
+			    // TAG Stuff
+			    $cat_type = 'forum post';
+			    $cat_objid = $_REQUEST["comments_threadId"];
+			    $cat_desc = substr($_REQUEST["comments_data"],0,200);
+			    $cat_name = $_REQUEST["comments_title"];
+			    $cat_href="tiki-view_forum_thread.php?comments_parentId=" . $_REQUEST["comments_threadId"] . "&forumId=" . $_REQUEST["forumId"];
+			    include_once ("freetag_apply.php");
+
 			    // PROCESS ATTACHMENT HERE
 			    if( $threadId && isset($_FILES['userfile1']) && is_uploaded_file($_FILES['userfile1']['tmp_name']) )
 			    {
@@ -627,6 +648,16 @@
     $smarty->assign('reported', $commentslib->get_num_reported($_REQUEST['forumId']));
 }
 
+if ($feature_freetags == 'y') {
+    $cat_type = 'forum post';
+    $cat_objid = $_REQUEST["comments_threadId"];
+    include_once ("freetag_list.php");
+    //If in preview mode get the tags from the form and not from database
+    if (isset($_REQUEST["comments_previewComment"])) {
+	$smarty->assign('taglist',$_REQUEST["freetag_string"]);
+    }
+}
+
 include_once("textareasize.php");
 
 if ($feature_forum_parse == "y") {
