--- tiki-blog_post.php	2006-09-03 13:32:31.000000000 -0300
+++ tiki-blog_post.php	2007-06-28 22:34:32.000000000 -0300
@@ -70,7 +70,12 @@
 if (isset($_REQUEST["postId"]) && $_REQUEST["postId"] > 0) {
 	// Check permission
 	$data = $bloglib->get_post($_REQUEST["postId"]);
-
+	
+	// freetags
+	$cat_type='blog post';
+	$cat_objid = $_REQUEST["postId"];
+	include_once ("freetag_list.php");
+	
 	// If the user owns the weblog then he can edit
 	if ($user && $user == $blog_data["user"]) {
 		$data["user"] = $user;
@@ -115,7 +120,11 @@
 
 	$post_images = $bloglib->get_post_images($postId);
 	$smarty->assign_by_ref('post_images', $post_images);
-}
+} else {
+    // freetags - lista de sugestoes
+    include_once ("freetag_list.php");
+}	
+
 
 $smarty->assign('preview', 'n');
 
@@ -170,10 +179,11 @@
 	$smarty->assign('preview', 'y');
 }
 
+
 // remove images (permissions!)
 if (isset($_REQUEST["save"]) || isset($_REQUEST['save_exit'])) {
-	include_once ("lib/imagegals/imagegallib.php");
 
+	include_once ("lib/imagegals/imagegallib.php");
 	$smarty->assign('individual', 'n');
 
 	if ($userlib->object_has_one_permission($_REQUEST["blogId"], 'blog')) {
@@ -237,14 +247,25 @@
 
 	$edit_data = $imagegallib->capture_images($edit_data);
 	$title = isset($_REQUEST['title']) ? $_REQUEST['title'] : '';
+	
 	if ($_REQUEST["postId"] > 0) {
 		$bloglib->update_post($_REQUEST["postId"], $_REQUEST["blogId"], $edit_data, $user, $title, $_REQUEST['trackback']);
+		$postid = $_REQUEST["postId"];
 	} else {
 		$postid = $bloglib->blog_post($_REQUEST["blogId"], $edit_data, $user, $title, $_REQUEST['trackback']);
 
 		$smarty->assign('postId', $postid);
 	}
 
+	// freetags
+	$cat_type='blog post';
+	$cat_objid = $postid;
+	$cat_desc = $_REQUEST["data"];
+	$cat_name = $_REQUEST["title"];
+	$cat_href = "tiki-view_blog_post.php?blogId=". $_REQUEST["blogId"] . "&postId=" . $postid;
+	include_once("freetag_list.php");
+	include_once("freetag_apply.php");
+
 	if (isset($_REQUEST['save_exit'])) {
 		header ("location: tiki-view_blog.php?blogId=$blogId");
 
@@ -308,6 +329,14 @@ 
 ask_ticket('blog');
 
+require_once("el-tags_ajax.php");
+if (isset($_REQUEST["freetag_string"])) {
+    $smarty->assign_by_ref('taglist', $_REQUEST["freetag_string"]);
+}
+
+$useJs=$tikilib->get_user_preference($user,'user_useEditJs');
+$smarty->assign('useJavascript',$useJs);
+
 // disallow robots to index page:
 $smarty->assign('metatag_robots', 'NOINDEX, NOFOLLOW');
 
