--- lib/tikilib.php.old	2006-07-10 22:47:17.000000000 -0300
+++ lib/tikilib.php	2006-07-10 22:50:07.000000000 -0300
@@ -5292,8 +5292,13 @@
 				$listate = substr($line, $listlevel, 1);
 				if (($listate == '+' || $listate == '-') && !($litype == '*' && !strstr(current($listbeg), '</ul>') || $litype == '#' && !strstr(current($listbeg), '</ol>'))) {
 				    $thisid = 'id' . microtime() * 1000000;
-				    $data .= '<br /><img class="flipper" id="flipper' . $thisid . '" class="pointer" onclick="flip(\'' . $thisid . '\');toggleImage(this,\'iArrowGrey' . ($listate == '-' ? 'Down' : 'Right') . '.png\')" src="styles/estudiolivre/iArrowGrey' . ($listate == '-' ? 'Right' : 'Down') . '.png">';
-				    $listyle = ' id="' . $thisid . '" style="display:' . ($listate == '+' ? 'block' : 'none') . ';"';
+				    if (preg_match("|tiki-print.php|",$_SERVER['REQUEST_URI'])){
+				    	// $data .= '<br /><img class="flipper" id="flipper' . $thisid . '" class="pointer" onclick="flip(\'' . $thisid . '\');toggleImage(this,\'iArrowGreyRight.png\')" src="styles/estudiolivre/iArrowGreyDown.png">';
+				    	$listyle = ' id="' . $thisid . '"';
+				    } else {
+				    	$data .= '<br /><img class="flipper" id="flipper' . $thisid . '" class="pointer" onclick="flip(\'' . $thisid . '\');toggleImage(this,\'iArrowGrey' . ($listate == '-' ? 'Down' : 'Right') . '.png\')" src="styles/estudiolivre/iArrowGrey' . ($listate == '-' ? 'Right' : 'Down') . '.png">';
+				    	$listyle = ' id="' . $thisid . '" style="display:' . ($listate == '+' ? 'block' : 'none') . ';"';
+				    }
 				    $addremove = 1;
 				}
 			    }
@@ -5307,6 +5312,9 @@
 			$listate = substr($line, $listlevel, 1);
 			if (($listate == '+' || $listate == '-')) {
 			    $thisid = 'id' . microtime() * 1000000;
+			    if (preg_match("|tiki-print.php|",$_SERVER['REQUEST_URI'])){
+			    	$listate = '-';
+			    }
 			    $data .= '<br /><img class="flipper" id="flipper' . $thisid . '" class="pointer" onclick="flip(\'' . $thisid . '\');toggleImage(this,\'iArrowGrey' . ($listate == '-' ? 'Down' : 'Right') . '.png\')" src="styles/estudiolivre/iArrowGrey' . ($listate == '-' ? 'Right' : 'Down') . '.png">';
 			    $listyle = ' id="' . $thisid . '" style="display:' . ($listate == '+' ? 'block' : 'none') . ';"';
 			    $addremove = 1;
@@ -5365,8 +5373,11 @@
 			if ($divstate == '+' || $divstate == '-') {
 			    // OK. Must insert flipper after HEADER, and then open new div...
 			    $thisid = 'id' . microtime() * 1000000;
-			    $aopen = '<img class="flipper" id="flipper' . $thisid . '" class="pointer" onclick="flip(\'' . $thisid . '\');toggleImage(this,\'iArrowGrey' . ($divstate == '-' ? 'Down' : 'Right') . '.png\')" src="styles/estudiolivre/iArrowGrey' . ($divstate == '-' ? 'Right' : 'Down') . '.png">';
-			    $aclose .= '<div id="' . $thisid . '" style="display:' . ($divstate == '+' ? 'block' : 'none') . ';">';
+			    if(preg_match("|tiki-print.php|",$_SERVER['REQUEST_URI'])){
+			    	$divstate = '+';
+			    }
+			    $aopen = '<span class="hiddenPointer" onclick="flip(\'' . $thisid . '\');toggleImage(document.getElementById(\'flipper' . $thisid . '\'),\'iArrowGrey' . ($divstate == '-' ? 'Down' : 'Right') . '.png\')"> <img class="flipper pointer" id="flipper' . $thisid . '" src="styles/estudiolivre/iArrowGrey' . ($divstate == '-' ? 'Right' : 'Down') . '.png">';
+			    $aclose .= '<div id="' . $thisid . '" style="display:' . ($divstate == '+' ? 'block' : 'none') . ';"> </span>';
 			    array_unshift($divdepth, $hdrlevel);
 			    $addremove = 1;
 			}
