--- db/tiki-db.php	2007-11-12 18:45:39.000000000 -0200
+++ db/tiki-db.php	2007-11-12 18:51:43.000000000 -0200
@@ -123,9 +123,13 @@
 if (!@$dbTiki->Connect($host_tiki, $user_tiki, $pass_tiki, $dbs_tiki)) {
 	print "
 <html><body>
+<center><img src=\"styles/bolha/img/logoTop.png\">
+<h2>Estamos em manuten&ccedil;&atilde;o</h2>
+<h3>Volte mais tarde</h3></center>
+<!-- 
 <h1><font color='red'>Unable to connect to the database !</font></h1>
 ";
-	print $dbTiki->ErrorMsg();
+	print $dbTiki->ErrorMsg() . "-->";
 	exit;
 }
 
