--- tiki-view_blog.php	2006-11-02 11:35:45.000000000 -0300
+++ tiki-view_blog.php	2007-06-28 21:56:10.000000000 -0300
@@ -10,6 +10,18 @@
 require_once ('tiki-setup.php');
 
 include_once ('lib/blogs/bloglib.php');
+
+global $feature_freetags;
+global $tiki_p_view_freetags;
+
+if ($feature_freetags == 'y' and $tiki_p_view_freetags == 'y') {
+    global $freetaglib;
+    if (!is_object($freetaglib)) {
+	include_once('lib/freetag/freetaglib.php');
+    }
+    global $freetaglib;
+}
+
 if ($feature_categories == 'y') {
 	global $categlib;
 	if (!is_object($categlib)) {
@@ -183,11 +195,22 @@
 
 // Get a list of last changes to the blog database
 $listpages = $bloglib->list_blog_posts($_REQUEST["blogId"], $offset, $blog_data["maxPosts"], $sort_mode, $find, $now);
-
 $temp_max = count($listpages["data"]);
+
 for ($i = 0; $i < $temp_max; $i++) {
+
 	$listpages["data"][$i]["parsed_data"] =preg_replace("/&#([0-9]+);/me", "chr('\\1')", strtr($tikilib->parse_data($bloglib->get_page($listpages["data"][$i]["data"], 1)), array_flip(get_html_translation_table(HTML_ENTITIES)))) ;
 
+	// freetags
+	$cat_type='blog post';
+	$cat_objid = $listpages["data"][$i]["postId"];
+
+	$tags = $freetaglib->get_tags_on_object($cat_objid, $cat_type);
+	$taglist = array();
+	for ($j=0; $j<sizeof($tags['data']); $j++) {
+	    $taglist[] = $tags['data'][$j]['tag'];
+	}
+	$listpages["data"][$i]["tags"] = $taglist;
 //print(htmlspecialchars($listpages["data"][$i]["parsed_data"]));
 }
 
