--- lib/tikilib.php	2006-06-13 22:23:24.000000000 -0300
+++ lib/tikilib.php	2006-06-13 22:22:49.000000000 -0300
@@ -5287,7 +5290,7 @@
 				$listate = substr($line, $listlevel, 1);
 				if (($listate == '+' || $listate == '-') && !($litype == '*' && !strstr(current($listbeg), '</ul>') || $litype == '#' && !strstr(current($listbeg), '</ol>'))) {
 				    $thisid = 'id' . microtime() * 1000000;
-				    $data .= '<br /><a id="flipper' . $thisid . '" class="link" href="javascript:flipWithSign(\'' . $thisid . '\')">[' . ($listate == '-' ? '+' : '-') . ']</a>';
+				    $data .= '<br /><img class="flipper" id="flipper' . $thisid . '" class="pointer" onclick="flip(\'' . $thisid . '\');toggleImage(this,\'iArrowGrey' . ($listate == '-' ? 'Down' : 'Right') . '.png\')" src="styles/estudiolivre/iArrowGrey' . ($listate == '-' ? 'Right' : 'Down') . '.png">';
 				    $listyle = ' id="' . $thisid . '" style="display:' . ($listate == '+' ? 'block' : 'none') . ';"';
 				    $addremove = 1;
 				}
@@ -5302,7 +5305,7 @@
 			$listate = substr($line, $listlevel, 1);
 			if (($listate == '+' || $listate == '-')) {
 			    $thisid = 'id' . microtime() * 1000000;
-			    $data .= '<br /><a id="flipper' . $thisid . '" class="link" href="javascript:flipWithSign(\'' . $thisid . '\')">[' . ($listate == '-' ? '+' : '-') . ']</a>';
+			    $data .= '<br /><img class="flipper" id="flipper' . $thisid . '" class="pointer" onclick="flip(\'' . $thisid . '\');toggleImage(this,\'iArrowGrey' . ($listate == '-' ? 'Down' : 'Right') . '.png\')" src="styles/estudiolivre/iArrowGrey' . ($listate == '-' ? 'Right' : 'Down') . '.png">';
 			    $listyle = ' id="' . $thisid . '" style="display:' . ($listate == '+' ? 'block' : 'none') . ';"';
 			    $addremove = 1;
 			}
@@ -5352,6 +5355,7 @@
 			// OK. Parse headers here...
 			$anchor = '';
 			$aclose = '';
+			$aopen = '';
 			$addremove = 0;
 
 			// May be special signs present after '!'s?
@@ -5359,7 +5363,7 @@
 			if ($divstate == '+' || $divstate == '-') {
 			    // OK. Must insert flipper after HEADER, and then open new div...
 			    $thisid = 'id' . microtime() * 1000000;
-			    $aclose = '<a id="flipper' . $thisid . '" class="link" href="javascript:flipWithSign(\'' . $thisid . '\')">[' . ($divstate == '-' ? '+' : '-') . ']</a>';
+			    $aopen = '<img class="flipper" id="flipper' . $thisid . '" class="pointer" onclick="flip(\'' . $thisid . '\');toggleImage(this,\'iArrowGrey' . ($divstate == '-' ? 'Down' : 'Right') . '.png\')" src="styles/estudiolivre/iArrowGrey' . ($divstate == '-' ? 'Right' : 'Down') . '.png">';
 			    $aclose .= '<div id="' . $thisid . '" style="display:' . ($divstate == '+' ? 'block' : 'none') . ';">';
 			    array_unshift($divdepth, $hdrlevel);
 			    $addremove = 1;
@@ -5377,7 +5381,7 @@
 			    $pageNumLink = ($pageNum >= 2)? "tiki-index.php?page=$page&pagenum=$pageNum": "";
 			    array_push($anch, str_repeat("*", $hdrlevel). " <a href='$pageNumLink#$thisid' class='link'>" . substr($line, $hdrlevel + $addremove). '</a>');
 			}
-			$line = $anchor . "<h$hdrlevel>" . substr($line, $hdrlevel + $addremove). "</h$hdrlevel>" . $aclose;
+			$line = $anchor . "<h$hdrlevel>" . $aopen . substr($line, $hdrlevel + $addremove). "</h$hdrlevel>" . $aclose;
 		    } elseif (!strcmp($line, "...page...")) {
 			// Close open paragraph, lists, and div's
 			$this->close_blocks($data, $in_paragraph, $listbeg, $divdepth, 1, 1, 1);
