--- lib/tikilib.php	2006-07-10 15:15:10.000000000 -0300
+++ lib/tikilib.php	2006-07-10 15:22:58.000000000 -0300
@@ -4240,7 +4240,7 @@
 		$attrib .= 'target="_blank" ';
 	if ($feature_wiki_ext_icon == 'y') {
 		$attrib .= 'class="wiki external" ';
-		$ext_icon = "<img border=\"0\" class=\"externallink\" src=\"img/icons/external_link.gif\" alt=\"external link\" />";
+		$ext_icon = "";
 	} else {
 		$attrib .= 'class="wiki" ';
 		$ext_icon = "";
@@ -4984,8 +4984,10 @@
 	    } else {
 	    	global $feature_wiki_ext_icon;
 	    	if ($feature_wiki_ext_icon == 'y') {
+		    if (preg_match("|^http://|",$link) && !preg_match("/estudiolivre\.org/", $link)) {
 		    	$class = 'class="wiki external"';
-		    	$ext_icon = "<img border=\"0\" class=\"externallink\" src=\"img/icons/external_link.gif\" alt=\"external link\" />";
+		    	$ext_icon = "";
+		    }
 	    	}
 	    }
 
