--- ../estudiolivre_base/lib/userslib.php	2006-02-09 22:30:32.000000000 -0200
+++ lib/userslib.php	2006-06-14 19:11:20.000000000 -0300
@@ -1982,6 +1982,22 @@
 	return $result;
     }
 
+    //this function sets ione user field
+    function set_user_field($name, $value) {
+
+	global $user;
+
+	$aUserPrefs = array('realName','site','local', 'isPublic', 'licencaPadrao', 'allowMsgs');
+	if (in_array($name, $aUserPrefs)) {
+	    $result = $this->set_user_preference($user, $name, $value);
+	} else {
+	    $query = "UPDATE `users_users` set `$name`='$value' where `login`='$user'";
+	    $result = $this->query($query);
+	}
+
+	return $result;
+    }
+
     // damian aka damosoft
     function count_users($group) {
         static $rv = array();
